x-pack/plugins/enterprise_search/public/applications/__mocks__/index.ts

```diff
@@ -9,7 +9,11 @@ export { mockKibanaValues } from './kibana_logic.mock';
 export { mockLicensingValues } from './licensing_logic.mock';
 export { mockHttpValues } from './http_logic.mock';
 export { mockTelemetryActions } from './telemetry_logic.mock';
-export { mockFlashMessagesValues, mockFlashMessagesActions } from './flash_messages_logic.mock';
+export {
+  mockFlashMessagesValues,
+  mockFlashMessagesActions,
+  mockFlashMessageHelpers,
+} from './flash_messages_logic.mock';
```

Comment:
I've never really understood the value of creating these kinds of `index.ts` files for bundling unrelated code in sub-directories.  That this introduces side-effects (if you import anything from this file _everything_ gets mocked) seems REALLY weird, and is not something I would expect to encounter, and I imagine would be gnarly to debug.  You mention a solution is to import these mocks individually in files, which you do for a couple of tests.  Maybe we should handle all mock imports that way? Is there an upside to this I'm not seeing that's worth the side-effect?

Reply:
Ya it's definitely an architectural decision. As with everything it's all about dev tradeoffs, in my personal opinion I do strongly feel that it's nice to not have to dive deeply into folders - when I'm outside that folder but reaching into it from another component, it's very nice to just have everything available from the top-level `index.ts` file and not to have 3 different lines of imports for 3 different files.

It's also one that Kibana uses everywhere ([example file](https://github.com/elastic/kibana/blob/master/src/core/public/index.ts) so I personally feel confident following them in this area. Should they ever change this (which I imagine would primarily be due to performance reasons, tree shaking, etc.), I would definitely be +1 to following suit.

I will also note that the side effects are limited to tests/`jest.mock()` behavior, whereas the benefits of quicker and combined imports apply outside those and to our source code as well (the main benefit IMO). For me at least the cost is well worth it once imports are understood - they were only unexpected to me because I didn't really understand how they worked. I strongly also think we'll find any side effects do not negatively impact like 95+% of our final code base.